### PR TITLE
AI core is now Abductor-proof.

### PR DESCRIPTION
**Pull request by Winterous**
## About The Pull Request

Simply prevents sending the teleport into the AI core.

## Why It's Good For The Game

Policy thread on the topic. https://tgstation13.org/phpBB/viewtopic.php?p=470229#p470229
Abductors not only having the means but strong incentive to immediately and near-unpreventably round-remove the AI rather than subverting them has long struck me as frankly unacceptable.
Their stated goal is to experiment and observe, they are explicitly not supposed to murder people, but the AI is one of the few things that actually has the power to make their life difficult; and they have free reign to dispose of one as they please.

There's no counterplay other than two things.
1. Metagaming and putting pingsky underneath your core ahead of time, and even that isn't foolproof, since there's a brief moment they can card you before he stuns them.
2. Cutting the cameras in your core (which is obviously not ideal) so they can't see the tiles to teleport to.

Even surrounding yourself with walls doesn't help, since they can teleport into solid matter somehow. It's probably the single worst-feeling way to get screwed in the game.

## Changelog
:cl:
balance: Station AI cores are now stored in a tin foil-lined room, in response to reports of mysterious teleporting AI thieves.
/:cl:

### CODE REVIEW

**Comment by Winterous** [comment]
Now, I'm very open to alternatives here, it's a bit of a heavy-handed shutdown but from my perspective it's a serious issue.
I think it should frankly be expected of abductors to abide by a catch-and-release method of operation, AIs and silicons included; drag them in, fiddle with their mind, send them out. But I'm new to coding and that would be a huge struggle to me.

All I know is no one seems to have changed it since I first complained like two and a half years ago, and figure it's about time something changed.

**Comment by Winterous** [comment]
I am incapable of interpreting what that Travis merge conflict is about. plz halp

File: code/modules/antagonists/abductor/machinery/camera.dm [hunk 1]
```diff
@@ -76,6 +76,11 @@
 	var/mob/living/carbon/human/C = owner
 	var/mob/camera/ai_eye/remote/remote_eye = C.remote_control
 	var/obj/machinery/abductor/pad/P = target
+
+	if(istype(get_area(remote_eye), /area/ai_monitored/turret_protected/ai))
```
Comment (MrDoomBringer) [inline code/modules/antagonists/abductor/machinery/camera.dm:80]: what? Don't we have `noteleport` flags for areas or something that does exactly this? Or does the abductor console bypass that?

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

**Comment by necromanceranne** [comment]
The AI doesn't even pose a threat to abductors in the first place, why did that policy thread imply otherwise. The vast majority of abductor gear renders AIs incredibly ineffectual in stopping abductors.

File: code/modules/antagonists/abductor/machinery/camera.dm (hunk 1 above)
Comment (Winterous) [reply code/modules/antagonists/abductor/machinery/camera.dm:80]: It bypasses, I swear I read something along those lines back in the day but I searched for protected and couldn't find it.

**Comment by ExcessiveUseOfCobblestone** [comment]
whats the AI gonna do, bolt them in lol?

File: code/modules/antagonists/abductor/machinery/camera.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/antagonists/abductor/machinery/camera.dm:80]: Noteleport also prevents teleportation though

**Comment by TheMidnightRose** [comment]
Also Cutting cameras does nothing. The AI itself is a camera.

**Comment by Tlaltecuhtli** [comment]
pr is just duct tape, add bluespace shielding machinery so we have actual content

**Comment by zxaber** [comment]
This would be better done by a machine in the room that prevents teleportation within it's area, rather than an unexplained magic no-teleport room. Could tie it to the power, so if power ever dies, the protection drops with it.

**Comment by ATH1909** [comment]
Aw, one of my favorite things to do as an abductor is card the AI and implant it in some rando's chest with a special lawset to help the person it's implanted in, but also keep its presence a secret.

**Comment by Winterous** [comment]
It's not unexplained magic teleport blocking, the AI's in a heavily reinforced area that their teleporter can't properly penetrate. Got wave-scrambling structures in the walls and shit. Something like that. Centcom blocking teleports uses the same reasoning.

**Comment by ATH1909** [comment]
> It's not unexplained magic teleport blocking, the AI's in a heavily reinforced area that their teleporter can't properly penetrate. Got wave-scrambling structures in the walls and shit. Something like that. Centcom blocking teleports uses the same reasoning.

Doesn't the AI sat literally have a teleporter beacon and a teleporter in it?

**Comment by MrDoomBringer** [comment]
@Winterous you're missing the point. When nabski says magic, he says that not because there isn't a great in-lore explanation for it, but because there isn't a physical, interactable in-game object that relates to the feature. Having a machine that determines if an area is 'no-teleport' opens up much more emergent gameplay and is in general much better game design than having the area intrinsically be impossible to teleport to. 

Centcom, of course, doesn't need something like this because it's divorced from normal gameplay. It can have "wave-scrambling structures in the walls and shit" - which for all intents and purposes boils down to 'magic anti-teleport zone'.

**Comment by optimumtact** [comment]
you need to prefix your changlog entry

**Comment by Winterous** [comment]
> 
> 
> > It's not unexplained magic teleport blocking, the AI's in a heavily reinforced area that their teleporter can't properly penetrate. Got wave-scrambling structures in the walls and shit. Something like that. Centcom blocking teleports uses the same reasoning.
> 
> Doesn't the AI sat literally have a teleporter beacon and a teleporter in it?

Yes it does, but that's a transmitter that gets locked on to, not an 'aimed' teleport.

And whatever it doesn't need to make sense. Why can't stray cargo pods land in the AI sat? Gameplay reasons, that's why!

If someone wants to code a more immersive and interesting alternative to this then be my guest, but apparently I'm the only person who cared enough to make this change, and this is what my skill level is capable of achieving.

**Comment by ExcessiveUseOfCobblestone** [comment]
Adding a sabotagable point validates the behavior that abductors should be grieffing the AI as an opportunity arises rather than to just focus on getting humans and conducting weird surgery though.

**Comment by TheMidnightRose** [comment]
If we want to go the route of 'machine that blocks teleports without tracking beacons' shove the machine in the Tcomms room and make it block telepads and briefcase telepads till either its depowered or destroyed. Though make it so that only protects rooms designated with a blueprint name. That way. It protects the AI core and can be a target by traitors and Abductors (If we allow abductors to steal AI still)

**Comment by actioninja** [comment]
wow thats a lot of bikeshedding

**Pull request merged**